### PR TITLE
ADD async render support

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: -add is sync but sometimes we want to do some async setup on first render-

## What I did
Add support for a function in `.add()` that returns a Promise

## Todo
- [ ] Add this to docs

## How to test
Should add something to the kitchen sink probably.

### CODE REVIEW

**Review by shilman — CHANGES_REQUESTED** [review]
Looks pretty straightforward. But needs:
 - [ ] docs including a motivating example
 - [ ] test case

**Review by danielduan — APPROVED** [review]

File: app/react/src/client/preview/render.js [hunk 1]
```diff
@@ -70,32 +70,32 @@ export function renderMain(data, storyStore) {
     story: selectedStory,
   };
 
-  const element = story(context);
-
-  if (!element) {
-    const error = {
-      title: `Expecting a React element from the story: "${selectedStory}" of "${selectedKind}".`,
-      description: stripIndents`
+  return Promise.resolve(story(context)).then(element => {
```
Comment (danielduan) [inline app/react/src/client/preview/render.js]: I think it'd be nice to have an example usage of this somewhere.

Comment (ndelangen) [inline app/react/src/client/preview/render.js]: Good point!

**Comment by ndelangen** [comment]
I'm dropping this for now, and will revisit this at a later stage.
#1209

**Comment by ndelangen** [comment]
Please do **NOT** remove remote branch

**Comment by swillis12** [comment]
Any plans to pick this back up? Looks like branch was deleted.

**Pull request closed without merging**